### DISCUSSION TITLE
Unwanted LostFocus when child ComboBox is selected

### DISCUSSION THREAD

**Discussion by cristinathoughtpennies** [AvaloniaUI/Avalonia#9869] [Q&A] [answered]
Hello!
I want to be able to take an action when a DockPanel containing a TextBlock and ComboBox has lost focus, so I am using an EventTriggerBehavior to do this.
The issue is that when the ComboBox is selected, this is registered as the Panel losing focus.
Is there a different event I should use or a different way to go about this, so that selecting the child ComboBox isn't considered as leaving the Parent Panel?

**Comment by robloo** [comment, marked as answer]
Only one UI element may be focused at a time. To select the ComboBox child will remove focus of the parent. I think you will need to re-think a bit how your logic works.

However, Avalonia has the `:focus-within` and `InputElement.IsKeyboardFocusWithin` properties which might have the information you need.

**Reply by cristinathoughtpennies** [reply]
I'll take a look at these, thanks!
